addons/html_builder/static/src/core/builder_selection_restriction_plugin.js

```diff
@@ -0,0 +1,242 @@
+                // edge case: when click the image, then ctrl+a the selection should stay on the image
+                editableSelection.commonAncestorContainer.nodeName === "FIGURE" ||
+                // edge case: when main body is empty, and click the footer outer blue container
+                // then ctrl+a, the selection is collapsed in editable but to the main body div,
+                // which causes options updating
+                (editableSelection.anchorNode.isContentEditable === false &&
+                    editableSelection.isCollapsed)
+            ) {
+                ev.preventDefault();
+                ev.stopPropagation();
+                return;
+            }
+            ev.preventDefault();
+            const closestSpecialBlock = closestElement(
+                editableSelection.anchorNode,
+                this.special_block_with_text_in_non_div.join(",")
+            );
+            const closestDiv = closestElement(editableSelection.anchorNode, "div");
+
+            // if the closest special block doesn't contains the closest div block,
+            // we select the paragraph, otherwise we select the whole closest div block
+            if (closestSpecialBlock && !closestSpecialBlock.contains(closestDiv)) {
+                this.selectAllInElement(closestElement(editableSelection.anchorNode, "p"));
+            } else {
+                this.selectAllInElement(closestDiv);
+            }
+            ev.stopPropagation();
+        }
+    }
+
+    // we extend the selection to the whole element step by step to properly
+    // handle uncrossable elements (like row, blockquote...)
+    selectAllInElement(element) {
+        let selection = this.dependencies.selection.getEditableSelection();
+        let { anchorNode, anchorOffset, focusNode, focusOffset, direction } = selection;
+
+        const [newAnchorNode, newAnchorOffset] = getDeepestPosition(element, 0);
+        const [newFocusNode, newFocusOffset] = getDeepestPosition(element, nodeSize(element));
+        if (direction === DIRECTIONS.RIGHT) {
+            this.dependencies.selection.setSelection({
+                anchorNode: focusNode,
+                anchorOffset: focusOffset,
+                focusNode: newAnchorNode,
+                focusOffset: newAnchorOffset,
+            });
+        } else {
+            this.dependencies.selection.setSelection({
+                anchorNode: anchorNode,
+                anchorOffset: anchorOffset,
+                focusNode: newAnchorNode,
+                focusOffset: newAnchorOffset,
+            });
+        }
+        this.correctSelectionOnUncrossable();
+        selection = this.dependencies.selection.getEditableSelection();
+        ({ focusNode, focusOffset } = selection);
+        this.dependencies.selection.setSelection({
+            anchorNode: focusNode,
+            anchorOffset: focusOffset,
+            focusNode: newFocusNode,
+            focusOffset: newFocusOffset,
+        });
```

Comment:
I know why you had to do `({ focusNode, focusOffset } = selection);`, but as it doesn't look good, maybe directly do `selection.focusNode` :slightly_smiling_face: